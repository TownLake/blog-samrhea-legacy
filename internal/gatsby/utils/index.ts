export { default as toKebabCase } from "./to-kebab-case";
export { default as decrement } from "./decrement";
export { default as increment } from "./increment";
export { default as concat } from "./concat";
