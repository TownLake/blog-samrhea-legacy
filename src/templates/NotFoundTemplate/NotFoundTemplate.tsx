import React from "react";

import { Layout } from "@/components/Layout";
import { Meta } from "@/components/Meta";
import { Page } from "@/components/Page";
import { Sidebar } from "@/components/Sidebar";
import { useSiteMetadata } from "@/hooks";

const NotFoundTemplate: React.FC = () => (
  <Layout>
    <Sidebar />
    <Page title="NOT FOUND">
      <p>You just hit a route that doesn't exist... the sadness.</p>
    </Page>
  </Layout>
);

export const Head: React.FC = () => {
  const { title, subtitle } = useSiteMetadata();
  const pageTitle = `Not Found - ${title}`;

  return <Meta title={pageTitle} description={subtitle} />;
};

export default NotFoundTemplate;
