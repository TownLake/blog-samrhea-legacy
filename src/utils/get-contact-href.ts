import { btoa } from "abab";

const getContactHref = (name: string, contact: string) => {
  const hrefs: { [key: string]: string } = {
    email: btoa(contact) || "",
    line: `line://ti/p/${contact}`,
    telegram: `https://t.me/${contact}`,
    vkontakte: `https://vk.com/${contact}`,
    medium: `https://medium.com/${contact}`,
    github: `https://github.com/${contact}`,
    weibo: `https://www.weibo.com/${contact}`,
    gitlab: `https://www.gitlab.com/${contact}`,
    codepen: `https://www.codepen.io/${contact}`,
    twitter: `https://www.twitter.com/${contact}`,
    soundcloud: `https://soundcloud.com/${contact}`,
    facebook: `https://www.facebook.com/${contact}`,
    instagram: `https://www.instagram.com/${contact}`,
    linkedin: `https://www.linkedin.com/in/${contact}`,
    youtube: `https://www.youtube.com/channel/${contact}`,
    mastodon: `${contact}`,
  };

  return hrefs[name] ?? contact;
};

export default getContactHref;
