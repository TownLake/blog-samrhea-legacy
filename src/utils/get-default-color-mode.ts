const getDefaultColorMode = (): "dark" | "light" => {
  if (typeof window === "undefined") {
    return "light";
  }

  const mql = window.matchMedia("(prefers-color-scheme: dark)");
  const hasMediaQueryPreference = typeof mql.matches !== "undefined";
  if (hasMediaQueryPreference) {
    return mql.matches ? "dark" : "light";
  }

  return "light";
};

export default getDefaultColorMode;
