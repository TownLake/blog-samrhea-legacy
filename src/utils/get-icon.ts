import { ICONS } from "@/constants";

const getIcon = (name: keyof typeof ICONS) => ICONS[name] || {};

export default getIcon;
