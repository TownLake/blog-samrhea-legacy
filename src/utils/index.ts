export { default as getIcon } from "./get-icon";
export { default as getContactHref } from "./get-contact-href";
export { default as toKebabCase } from "./to-kebab-case";
export { default as getDefaultColorMode } from "./get-default-color-mode";
export * as testUtils from "./test-utils";
