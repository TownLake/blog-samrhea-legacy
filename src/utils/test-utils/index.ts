export {
  default as renderWithCoilProvider,
  createSnapshotsRenderer,
} from "./render-with-coil-provider";
export { default as getMeta } from "./get-meta";
