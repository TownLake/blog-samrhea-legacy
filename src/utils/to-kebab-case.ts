const toKebabCase = (str: string = ""): string =>
  str
    .match(/[A-Z]{2,}(?=[A-Z][a-z]+[0-9]*|\b)|[A-Z]?[a-z]+[0-9]*|[A-Z]|[0-9]+/g)
    ?.map((word) => word.toLowerCase())
    .join("-") || "";

export default toKebabCase;
