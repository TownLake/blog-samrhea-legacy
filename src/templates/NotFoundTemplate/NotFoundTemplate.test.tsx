import React from "react";

import { render as reactTestingLibraryRender } from "@testing-library/react";
import { StaticQuery, useStaticQuery } from "gatsby";

import * as mocks from "@/mocks";
import { testUtils } from "@/utils";

import NotFoundTemplate, { Head as GatsbyHead } from "./NotFoundTemplate";

const mockedStaticQuery = StaticQuery as jest.Mock;
const mockedUseStaticQuery = useStaticQuery as jest.Mock;

describe("NotFoundTemplate", () => {
  beforeEach(() => {
    mockedStaticQuery.mockImplementationOnce(({ render }) =>
      render(mocks.siteMetadata),
    );
    mockedUseStaticQuery.mockReturnValue(mocks.siteMetadata);
  });

  test("renders correctly", () => {
    const tree = testUtils
      .createSnapshotsRenderer(<NotFoundTemplate />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("head renders correctly", () => {
    reactTestingLibraryRender(<GatsbyHead />);

    expect(testUtils.getMeta("twitter:card")).toEqual("summary_large_image");
    expect(testUtils.getMeta("twitter:title")).toEqual(
      "Not Found - Blog by John Doe",
    );
    expect(testUtils.getMeta("og:title")).toEqual(
      "Not Found - Blog by John Doe",
    );
    expect(testUtils.getMeta("description")).toEqual(
      "Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu.",
    );
    expect(testUtils.getMeta("twitter:description")).toEqual(
      "Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu.",
    );
    expect(testUtils.getMeta("og:description")).toEqual(
      "Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu.",
    );
  });
});
