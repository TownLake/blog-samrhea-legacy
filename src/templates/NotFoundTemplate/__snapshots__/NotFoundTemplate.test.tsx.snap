// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundTemplate renders correctly 1`] = `
<div>
  <div>
    <div>
      <div>
        <a
          href="/"
        />
        <div>
          <h2>
            <a
              href="/"
            >
              John Doe
            </a>
          </h2>
        </div>
        <p>
          Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu.
        </p>
      </div>
      <nav>
        <ul>
          <li>
            <a
              href="/"
            >
              Articles
            </a>
          </li>
          <li>
            <a
              href="/pages/about"
            >
              About Me
            </a>
          </li>
          <li>
            <a
              href="/pages/contacts"
            >
              Contact Me
            </a>
          </li>
        </ul>
      </nav>
      <div>
        All rights reserved.
      </div>
    </div>
  </div>
  <div>
    <div>
      <h1>
        NOT FOUND
      </h1>
      <div>
        <p>
          You just hit a route that doesn't exist... the sadness.
        </p>
      </div>
    </div>
  </div>
</div>
`;
